src/lib/extension-load-queue.ts

```diff
@@ -0,0 +1,118 @@
+import {useEffect, useState} from 'react';
+
+declare global {
+    interface Window {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        [key: symbol]: any;
```

Comment:
This is still sub-optimal, even though I won't mind as much if you decide to leave this as is. An optimal typing strategy would be something like https://github.com/diplodoc-platform/html-extension/blob/dea1790907586c976d0ec3cf1a7f8dae7f1859ea/src/types.ts#L28

Reply:
In the example given, GLOBAL_SYMBOL is a pre-known constant. In mine, it is a key that is not known in advance. For the same reason, ScriptStore is generic in this PR. I propose to solve the problem of improved typing through hotel PR.